docs: Edge Function docs (#18738)
supabase/supabase: apps/docs/pages/guides/functions/auth.mdx

  id: 'auth',
  title: 'Integrating With Supabase Auth',
  description: 'Supabase Edge Functions and Auth.',
  subtitle: 'Supabase Edge Functions and Auth.',
}

Edge Functions work seamlessly with [Supabase Auth](/docs/guides/auth).

## Auth Context

When a user makes a request to an Edge Function, we can use the Authorization header to set the Auth context in the Supabase client:

```js mark=10
import { serve } from 'https://deno.land/std@0.177.0/http/server.ts'
import { createClient } from 'https://esm.sh/@supabase/supabase-js@2'

serve(async (req: Request) => {

  const authHeader = req.headers.get('Authorization')!
  const supabaseClient = createClient(
    Deno.env.get('SUPABASE_URL') ?? '',
    Deno.env.get('SUPABASE_ANON_KEY') ?? '',
    { global: { headers: { Authorization: authHeader } } }
  )

})
```

> Importantly, this is done _inside_ the `serve()` function, so that the Authorization header is set for each request.

Reviewer: ```suggestion
Importantly, this is done _inside_ the `Deno.serve()` callback argument, so that the Authorization header is set for each request.
```

## Fetching the user

> After initializing a Supabase client with the Auth context, we can use `getUser()` to fetch the user object, and run queries in the context of the user with [Row Level Security (RLS)](/docs/guides/auth/row-level-security) policies enforced.

Reviewer: ```suggestion
After initializing a Supabase client with the Auth context, you can use `getUser()` to fetch the user object, and run queries in the context of the user with [Row Level Security (RLS)](/docs/guides/auth/row-level-security) policies enforced.
```

```js mark=13:14

> import { serve } from 'https://deno.land/std@0.177.0/http/server.ts'
> import { createClient } from 'https://esm.sh/@supabase/supabase-js@2'
> 
> serve(async (req: Request) => {

Reviewer: ```suggestion
import { createClient } from 'https://esm.sh/@supabase/supabase-js@2'

Deno.serve(async (req: Request) => {
```

Not part of this PR I know, but need to fix it at the same time if we're changing above

  const supabaseClient = createClient(
    Deno.env.get('SUPABASE_URL') ?? '',
    Deno.env.get('SUPABASE_ANON_KEY') ?? '',
    { global: { headers: { Authorization: req.headers.get('Authorization')! } } }
  )

>   // Now we can get the session or user object

Reviewer: ```suggestion
  // Get the session or user object
```

  const { data } = await supabaseClient.auth.getUser()
  const user = data.user

  return new Response(JSON.stringify({ user }), {
    headers: { 'Content-Type': 'application/json' },
    status: 200,
  })

})
```

## Row Level Security

After initializing a Supabase client with the Auth context, all queries will be executed with the context of the user. For database queries, this means [Row Level Security](/docs/guides/auth/row-level-security) will be enforced.

```js mark=13

> import { serve } from 'https://deno.land/std@0.177.0/http/server.ts'
> import { createClient } from 'https://esm.sh/@supabase/supabase-js@2'
> 
> serve(async (req: Request) => {

Reviewer: ```suggestion
import { createClient } from 'https://esm.sh/@supabase/supabase-js@2'

Deno.serve(async (req: Request) => {
```
